tests/integration/ha/test_ha.py

```diff
@@ -29,6 +43,21 @@ async def test_deploy_active(ops_test: OpsTest):
     await helpers.wait_idle(ops_test)
 
 
+async def test_replication(ops_test: OpsTest):
```

Comment:
**question:** Do we need this test? We test replication in basically every other test already (although it's not explicit).

If we want a stand-alone replication test, I'd use the same continuous_writes set-up we have already, but writing ONLY to the leader, and looping over the non-leaders.
Something like:
```python
cw.start_continuous_writes(parent=parent, hosts=leader_host, username=USERNAME, password=password)
```
and...
```python
for host in non_leader_hosts.split(","):
    last_write_non_leader = cw.get_last_znode(
        parent=parent, hosts=host, username=USERNAME, password=password
    )
    total_writes_non_leader = cw.count_znodes(
        parent=parent, hosts=host, username=USERNAME, password=password
    )
    assert last_write == last_write_non_leader
    assert total_writes == total_writes_non_leader
```

Reply:
As an FYI, if you provide Kazoo with hosts `1,2,3`, it randomly picks one to **actually** write to, as Kazoo is ignorant to the current leader by default IIRC.

So if you want to check writes on a specific server, we need to pass a single host.

Reply:
This is the basic test on replication. It might be redundant, but it can be seen as a sanity check I guess, before moving forward with the rest of the test suite.

As per the zookeeper node to write to, the test already passes a single host, and then checks on all of them (also using a single host at a time)

Reply:
I agree, keeping it a sanity test is OK.

I would want us to use the continuous writes though, and loop through all the non-leader units to confirm that they're picking up ALL the writes, not just 1.